src/integrator.jl

```diff
@@ -523,5 +527,35 @@ function build_subintegrator_tree_with_cache(
     )
 end
 
+function build_subintegrator_tree_with_cache(
+    prob::OperatorSplittingProblem, alg::SciMLBase.AbstractODEAlgorithm,
+    f::F, p::P,
+    uprevouter::S, uouter::S,
+    solution_indices,
+    t0::T, dt::T, tf::T,
+    tstops, saveat, d_discontinuities, callback,
+    adaptive, verbose,
+    save_end=false,
+    controller=nothing,
+) where {S, T, P, F}
+    uprev = @view uprevouter[solution_indices]
+    u = @view uouter[solution_indices]
+
+    integrator = DiffEqBase.__init(
+        SciMLBase.ODEProblem(f, u, (t0, tf), p),
```

Comment:
This is tricky and I discarded this approach in the past, as this has some implications: How exactly do we know that this should *always* be an ODEProblem? I.e. do we have the guarantee that the combination `SciMLBase.AbstractODEAlgorithm` with any function type which we feed in is always an ODEProblem?

Another problem which I ran into was that I could not figure out how to connect the caches of the different integrators and how to control the used vector types (e.g. we sometimes want to solve part of the problem on the GPU and part on the CPU or some other device). See also that uprev is not passed into the integrator and hence we have allocations which we should be able to eliminate.

Is there any chance that we can address the points above if we follow this approach here?

Reply:
And for later, this should be something like
```suggestion
        SciMLBase.ODEProblem(f, u, (t0, min(t0+dt,tf)), p),
```
and adjusted for the correct time direction.